api/admin/controller/__init__.py

```diff
@@ -1740,16 +1740,24 @@ def custom_list(self, list_id):
             # Build the list of affected lanes before modifying the
             # CustomList.
             affected_lanes = Lane.affected_by_customlist(list)
+            surviving_lanes = []
             for lane in affected_lanes:
-                lane.update_size(self._db)
-                # There's only one custom list in the lane and it's going
-                # to be deleted.
-                if lane.size == 0 and len(lane._customlist_ids) == 1:
```

Comment:
There's a bug in this line exposed by my core changes. Even for a Lane based solely on one custom list, the lane won't be empty until after the list is deleted. The test was working because `update_size` used to use the materialized view, which wasn't being updated. Now it uses the database, which gives the right answer, so test lists weren't being deleted when they should have been.